vue/test/components/scrollable-items/scrollable-items.test.js

```diff
@@ -0,0 +1,394 @@
+        assert.strictEqual(component.find(".button-icon-previous").exists(), true);
+        assert.strictEqual(component.find(".button-icon-next").exists(), true);
+    });
+
+    it("should auto hide the arrows buttons", async () => {
+        const component = base.getComponent("ScrollableItems", {
+            props: {
+                items: [
+                    { label: "Item Example 1", value: "item1" },
+                    { label: "Item Example 2", value: "item2" },
+                    { label: "Item Example 3", value: "item3" },
+                    { label: "Item Example 4", value: "item4" },
+                    { label: "Item Example 5", value: "item5" },
+                    { label: "Item Example 6", value: "item6" }
+                ]
+            }
+        });
+
+        // Setting "scrollableWidth" greater than 0 is a workaround to make the test work properly.
+        // It makes the component think it doesn't have enough space to show all the items, thus
+        // the default behaviour will be to show the arrows
+        await component.setData({ scrollableWidth: 500 });
+        assert.strictEqual(component.find(".button-icon-previous").exists(), true);
+        assert.strictEqual(component.find(".button-icon-next").exists(), true);
+
+        await component.setData({ scrollableWidth: 0 });
+        assert.strictEqual(component.find(".button-icon-previous").exists(), false);
+        assert.strictEqual(component.find(".button-icon-next").exists(), false);
+
+        await component.setData({ scrollableWidth: 500 });
+        assert.strictEqual(component.find(".button-icon-previous").exists(), true);
+        assert.strictEqual(component.find(".button-icon-next").exists(), true);
+
+        await component.setProps({ autoHideArrows: false });
+        await component.setData({ scrollableWidth: 500 });
+        assert.strictEqual(component.find(".button-icon-previous").exists(), true);
+        assert.strictEqual(component.find(".button-icon-next").exists(), true);
+
+        await component.setData({ scrollableWidth: 0 });
+        assert.strictEqual(component.find(".button-icon-previous").exists(), true);
+        assert.strictEqual(component.find(".button-icon-next").exists(), true);
+    });
+
+    it("should change the selected item when it's clicked", async () => {
+        const component = base.getComponent("ScrollableItems", {
+            props: {
+                items: [
+                    { label: "Item Example 1", value: "item1" },
+                    { label: "Item Example 2", value: "item2" },
+                    { label: "Item Example 3", value: "item3" },
+                    { label: "Item Example 4", value: "item4" },
+                    { label: "Item Example 5", value: "item5" },
+                    { label: "Item Example 6", value: "item6" }
+                ]
+            },
+            mixins: [{ methods: { snapSelectedToCenter() {} } }]

```

Comment:
Comment on why this mock is needed.

Reply:
Better yet, mock it in `vue/test/base.js` with something like `window.HTMLElement.prototype.scrollIntoView = () => undefined;` so we don't have to repeat this for all tests (not only in this component but also in others).